src/main/java/org/spongepowered/api/world/extent/EntityUniverse.java

```diff
@@ -201,4 +205,207 @@
+    }
+
+    /**
+     * Gets all the entities that intersect the bounding box, in no particular
+     * order, as long as the pass the given filter test.
+     *
+     * @param box The intersection box
+     * @param filter The filter test
+     * @return All the intersecting entities that pass the filter test
+     */
+    Set<Entity> getIntersectingEntities(AABB box, Predicate<Entity> filter);
+
+    /**
+     * Gets all the entities that intersect the ray (by their bounding box)
+     * The ray is defined by its start and end point.
+     *
+     * @param start The start of the ray
+     * @param end The end of the ray
+     * @return The intersecting entities in no particular order, with the
+     * associated intersection point and normal
+     */
+    default Set<EntityHit> getIntersectingEntities(Vector3d start, Vector3d end) {
+        return getIntersectingEntities(start, end, hit -> true);
+    }
+
+    /**
+     * Gets all the entities that intersect the ray (by their bounding box)
+     * The ray is defined by its start and end point. Only the entities that
+     * pass the filter test are added.
+     *
+     * @param start The start of the ray
+     * @param end The end of the ray
+     * @param filter The filter test
+     * @return The intersecting entities in no particular order, with the
+     * associated intersection point and normal
+     */
+    Set<EntityHit> getIntersectingEntities(Vector3d start, Vector3d end, Predicate<EntityHit> filter);
+
+    /**
+     * Gets all the entities that are in the line of sight of the given entity,
+     * up to a given distance. This ignores occluders like blocks or other
+     * entities. That is to say, the returned entities might not actually be
+     * visible.
+     *
+     * @param looker The looking entity
+     * @param distance The distance of the ray (from the start)
+     * @return The intersecting entities in no particular order, with the
+     * associated intersection point and normal
+     */
+    default Set<EntityHit> getIntersectingEntities(Entity looker, double distance) {
+        return getIntersectingEntities(looker, distance, hit -> true);
+    }
+
+    /**
+     * Gets all the entities that are in the line of sight of the given entity,
+     * up to a given distance. This ignores occluders like blocks or other
+     * entities. That is to say, the returned entities might not actually be
+     * visible. Only the entities that pass the filter test are added.
+     *
+     * @param looker The looking entity
+     * @param distance The distance of the ray (from the start)
+     * @param filter The filter test
+     * @return The intersecting entities in no particular order, with the
+     * associated intersection point and normal
+     */
+    default Set<EntityHit> getIntersectingEntities(Entity looker, double distance, Predicate<EntityHit> filter) {
+        final Vector3d rotation = looker.getRotation();
```

Comment:
IMO this method should either use Living in the first place because  dead objects have no sight. or it should not deal with living and eye locations at all.